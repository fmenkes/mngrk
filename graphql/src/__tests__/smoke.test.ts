import {
  createTestClient,
  ApolloServerTestClient,
} from 'apollo-server-testing';
import { gql } from 'apollo-server';
import server from '../index';

let testClient: ApolloServerTestClient;

beforeAll(() => {
  testClient = createTestClient(server);
});

it('server starts up correctly', async () => {
  const { query } = testClient;

  const TEST_QUERY = gql`
    query IntrospectionQuery {
      __schema {
        mutationType {
          name
          description
        }
        queryType {
          name
          description
        }
      }
    }
  `;

  const res = await query({
    query: TEST_QUERY,
  });

  expect(res).toMatchSnapshot();
});
