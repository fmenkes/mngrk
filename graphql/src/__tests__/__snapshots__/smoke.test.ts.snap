// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server starts up correctly 1`] = `
Object {
  "data": Object {
    "__schema": Object {
      "mutationType": Object {
        "description": "",
        "name": "Mutation",
      },
      "queryType": Object {
        "description": "",
        "name": "Query",
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
